src/artm/regularizer/net_plsa_phi.cc

```diff
@@ -0,0 +1,156 @@
+// Copyright 2017, Additive Regularization of Topic Models.
+
+// Author: Murat Apishev (great-mel@yandex.ru)
+
+#include "artm/core/protobuf_helpers.h"
+#include "artm/core/phi_matrix.h"
+#include "artm/core/phi_matrix_operations.h"
+#include "artm/regularizer/net_plsa_phi.h"
+
+namespace artm {
+namespace regularizer {
+
+bool NetPlsaPhi::RegularizePhi(const ::artm::core::PhiMatrix& p_wt,
+                               const ::artm::core::PhiMatrix& n_wt,
+                               ::artm::core::PhiMatrix* result) {
+  // read the parameters from config and control their correctness
+  const int topic_size = p_wt.topic_size();
+  const int token_size = p_wt.token_size();
+
+  std::vector<bool> topics_to_regularize;
+  if (config_.topic_name().size() == 0) {
+    topics_to_regularize.assign(topic_size, true);
+  } else {
+    topics_to_regularize = core::is_member(p_wt.topic_name(), config_.topic_name());
+  }
+
+  if (!config_.has_class_id()) {
+    LOG(ERROR) << "There's no name of vertex modality in class_id field for" <<
+                  "NetPLSA regularizer. Cancel it's launch.";
+    return false;
+  }
+  const auto& class_id = config_.class_id();
+
+  bool has_weights = config_.vertex_weight_size();
+  if (has_weights && vertex_name2id_.size() != config_.vertex_weight_size()) {
+    LOG(ERROR) << "Non-empty vertex_weight array should have the same length " <<
+                  "with vertex_name array in NetPLSA regularizer config (" <<
+                  vertex_name2id_.size() << " != " << config_.vertex_weight_size() << ")";
+  }
+
+  auto normalizers = artm::core::PhiMatrixOperations::FindNormalizers(n_wt);
+  auto norm_iter = normalizers.find(class_id);
+  if (norm_iter == normalizers.end()) {
+    LOG(ERROR) << "NetPlsaPhiConfig.class_id " << class_id
+               << " does not exists in n_wt matrix. Cancel regularization.";
+  }
+  const auto& n_t = norm_iter->second;
+
+  for (int token_id = 0; token_id < token_size; ++token_id) {
+    const auto& token_u = p_wt.token(token_id);
+    if (token_u.class_id != class_id) {
+      continue;
+    }
+
+    auto name_iter = vertex_name2id_.find(token_u.keyword);
+    if (name_iter == vertex_name2id_.end()) {
```

Comment:
Is there any way to iterate not over Phi Matrix and search tokens in `vertex_name2id`, but to iterate over vertices, specified in config, and search corresponding tokens in Phi Matrix? It can be useful in case when we have much more tokens in Phi Matrix rather than in regularizer's config.